service/topology_coordinator.cc

```diff
@@ -1577,7 +1577,30 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         rtlogger.info("entered `{}` transition state", *tstate);
         switch (*tstate) {
             case topology::transition_state::join_group0: {
-                auto [node, accepted] = co_await finish_accepting_node(get_node_to_work_on(std::move(guard)));
```

Comment:
> It is important to do so before the
> new node start to get raft snapshot since the snapshot application is
> not atomic and queries that run parallel with it may see partial state
> and fail in weird ways.

If this is true, then we have the same problem for snapshot transfers in other contexts -- e.g. when a node is catching up after network partition. Correct?

Reply:
Of course. That is why we are working on making snapshot loading atomic. But what saves us here is that when we do tablets operations the topology coordinator makes sure that all nodes receive all relevant changes before starting to use an individual tablet. Here we break the assumption. At least this is how I convince myself that the problem is local and not that everything is completely broken.